File: frontend/src/types/video.ts
export enum VideoStatus {
  UPLOADING = "UPLOADING",
  UPLOADED = "UPLOADED",
  PROCESSING = "PROCESSING",
  READY = "READY",
  ERROR = "ERROR",
}

export interface Video {
  id: string;
  title: string;
  description?: string;
  status: VideoStatus;
  duration?: number;
  originalFilename: string;
  minioPath?: string;
  hlsPath?: string;
  fileSize: number;
  createdAt: string;
  updatedAt: string;
  owner: {
    id: string;
    username: string;
    fullName: string;
  };
}

export interface CreateVideoRequest {
  title: string;
  description?: string;
  file: File;
}

export interface VideoUploadProgress {
  videoId: string;
  progress: number;
  status: VideoStatus;
}

File: spring-video/src/main/java/com/didan/streaming/video/entity/Video.java
package com.didan.streaming.video.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "videos")
public class Video {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String title;
    private String description;
    private String originalFilename;
    private String minioPath;
    private String hlsPath;
    private Long fileSize;
    private Integer duration;

    @Enumerated(EnumType.STRING)
    private VideoStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id")
    private User owner;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
